### PR TITLE
Added negative integration tests for configuration sad paths.

### PR DESCRIPTION
**Pull request by knoxy5467**
This pull request adds a comprehensive suite of negative configuration tests for the Go integration tests of the Glue Schema Registry (GSR) Protobuf serializer and deserializer. It introduces new test cases to ensure that the system fails gracefully and as expected when provided with invalid or non-existent configuration values. The changes include both the new test logic and the necessary configuration files for various negative scenarios.

Negative configuration testing improvements:

* Added a new test suite in `configuration_test.go` that verifies the serializer and deserializer correctly handle missing configuration paths, invalid roles, non-existent registries, endpoints, and regions, ensuring robust error handling for misconfigurations.
* Created new configuration files for negative test scenarios, including invalid role to assume (`invalid_role_to_assume.properties`), non-existent registry (`non_existant_registry.properties`), non-existent endpoint (`non_existant_endpoint.properties`), and non-existent region (`non_existant_region.properties`). [[1]](diffhunk://#diff-1142a3ebdc2bf65e7ee9458f7ddcdcca43b2aca11ae1b9de43ce4ca7a741d7fbR1-R10) [[2]](diffhunk://#diff-6f6ac1cb3d3c014c92c174c87ef4b20f28d1688c7ba1369196441bb58f3cbde7R1-R9) [[3]](diffhunk://#diff-444db9a0abe7244805030909aeb5ad505e3d766704264fae529096297be1cffdR1-R9) [[4]](diffhunk://#diff-5890b8d3edb9d2365d1e50c46779f8d8a8fccf01847fae4f714575da048dae81R1-R9)

### CODE REVIEW

File: native-schema-registry/golang/integration-tests/gsr.properties [hunk 1]
```diff
@@ -6,5 +6,4 @@ endpoint=https://glue.us-east-1.amazonaws.com
 registry.name=default-registry
 description=Multi-threaded Protobuf GSR Kafka Demo with 5 message types
 compatibility=FORWARD
-dataFormat=PROTOBUF
```
Comment (isaurab007) [inline native-schema-registry/golang/integration-tests/gsr.properties]: I am tyring to understand why this is not necessary. Are we relying on it defaulting to a particular data type in the java layer?

File: native-schema-registry/golang/integration-tests/negative_test/configuration_test.go [hunk 2]
```diff
@@ -0,0 +1,279 @@
...
+
+	config := c.createProtobufConfig(MISSING_PATH)
+	phone := c.createTestPhone()
+
+	// Attempt to create serializer with missing config path
+	gsrSerializer, err := serializer.NewSerializer(config)
+	if err != nil {
+		// Expected: serializer creation should fail
+		c.T().Logf("✓ Serializer creation failed as expected: %v", err)
+		require.Error(c.T(), err, "Serializer creation should fail with missing config path")
```
Comment (isaurab007) [inline native-schema-registry/golang/integration-tests/negative_test/configuration_test.go:63]: Do we have any checks for the message string in error type?

**Comment by knoxy5467** [comment]
It is not being picked up by the underlying native Java. Therefore it has
no bearing.

File: native-schema-registry/golang/integration-tests/gsr.properties (hunk 1 above)
Comment (knoxy5467) [reply native-schema-registry/golang/integration-tests/gsr.properties]: This part of the configuration is not being picked up in the java layer. If I have it as PROTOBUF avro and jsonschema still work as expected.

Comment (isaurab007) [reply native-schema-registry/golang/integration-tests/gsr.properties]: Can you please post high level reason why this is happening? I thought java needs to be aware of the dataFormat when creating schemas. Have you tried after clearing up your registry of any previously created schemas?

File: native-schema-registry/golang/integration-tests/negative_test/configuration_test.go (hunk 2 above)
Comment (knoxy5467) [reply native-schema-registry/golang/integration-tests/negative_test/configuration_test.go:63]: are you saying that we want to validate that the message string should look like?
We can add checks to ensure the error code is 5005
